backends/graphs/controls.cpp

```diff
@@ -0,0 +1,304 @@
+/*
+Copyright 2013-present Barefoot Networks, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "config.h"
+
+#ifdef HAVE_LIBBOOST_GRAPH
+
+#include <boost/graph/graphviz.hpp>
+
+#include <iostream>
+
+#include "frontends/p4/methodInstance.h"
+#include "frontends/p4/tableApply.h"
+#include "lib/log.h"
+#include "lib/nullstream.h"
+#include "lib/path.h"
+
+#include "controls.h"
+
+namespace graphs {
+
+class EdgeUnconditional : public EdgeTypeIface {
+ public:
+    EdgeUnconditional() = default;
+    std::string label() const override { return ""; };
```

Comment:
changing to use `cstring` consistently; guess it makes sense since all the strings I use are pretty much meant to be immutable